zap/src/irgen/ser.rs

```diff
@@ -201,6 +201,34 @@ impl Ser {
 				));
 			}
 
+			Ty::Set(key) => {
+				let (len_name, len_expr) = self.add_occurrence("len");
+				let (len_pos_name, len_pos_expr) = self.add_occurrence("len_pos");
+
+				self.push_local(len_pos_name.clone(), Some(Var::from("alloc").call(vec![2.0.into()])));
+				self.push_local(len_name.clone(), Some(0.0.into()));
+
+				let (key_name, _) = self.add_occurrence("k");
+				let (val_name, _) = self.add_occurrence("_");
```

Comment:
Does `_` get picked up for shadowing and need an `add_occurrence`?

Reply:
Maybe it is good practice to keep it with `add_occurrence`?

Reply:
It becomes `__1`. Selene doesn't care about shadowing `_` but you're not writing normal Luau so there is no warning if you use an `_` variable. I'm fine with whichever, I could also add string splitting back to emit `_1`.

Reply:
Let's keep it as is then. No need to make it emit `_1` I reckon.